src/os/bluestore/KernelDevice.cc

```diff
@@ -231,21 +224,23 @@ int KernelDevice::collect_metadata(const string& prefix, map<string,string> *pm)
       break;
     default:
       {
+	char buffer[1024] = {0};
 	(*pm)[prefix + "partition_path"] = string(partition_path);
 	(*pm)[prefix + "dev_node"] = string(dev_node);
-	(*pm)[prefix + "model"] = get_dev_property(dev_node, BLKDEV_PROP_MODEL);
-	(*pm)[prefix + "dev"] = get_dev_property(dev_node, BLKDEV_PROP_DEV);
 
+	block_device_model(dev_node, buffer, sizeof(buffer));
+	(*pm)[prefix + "model"] = buffer;
+	
+	buffer[0] = '\0';
```

Comment:
`get_block_device_string_property()` already sets the ending `\0`.

Reply:
Not if block_device_wholedisk fails.